app/models/visualization.rb

```diff
@@ -3,4 +3,68 @@ class Visualization < ApplicationRecord
   serialize :data, JSON
   has_and_belongs_to_many :samples
   belongs_to :user
+
+  def samples_count
+    if visualization_type == "phylo_tree"
+      # WARNING: this could be an N+1 query
+      # TODO: (gdingle): store at write time if it is immutable
+      Sample.where(project: phylo_tree.project).count
+    else
+      samples.count
+    end
+  end
+
+  def phylo_tree
+    PhyloTree.find(data["treeId"])
+  end
+
+  def name
+    if visualization_type == "phylo_tree"
+      # WARNING: this could be an N+1 query
+      # TODO: (gdingle): store name at write time if it is immutable
+      phylo_tree.name
+    elsif samples.length == 1
+      samples[0].name
+    elsif samples.length > 1
+      names = samples.map(&:name)
+      # make a string such as "Patient 016 (CSF) and 015 (CSF)"
+      prefix = longest_common_prefix(names)
+      if prefix.include?(' ')
+        # Use whole words only
+        # TODO: (gdingle): add other delimiters than space
+        prefix = prefix.split(' ')[0..-2].join(' ')
```

Comment:
The problem here is that the prefix may include partial words such "patient 00" making "patient 001 and 2" when I think it's more natural to write "patient 001 and 002" . I think I will change `longest_common_prefix` to fix the root cause.

Reply:
> > Did you discuss this naming strategy with Jenn?
> 
> No, that's a good idea. @happyimadesignr: I came up with an auto naming algorithm for showing the heatmaps in the visualization list. It will makes a string such as "Patient 016 (CSF) and 015 (CSF)" from the two samples "Patient 016 (CSF)", "Patient 015 (CSF)". What do you think?
> 
> > Did you run a test phylo tree?
> 
> I ran it but it never seems to get past: "Computation in progress. Please check back later!". Same in master. Can I test in staging or is the a better way?

Yeah, staging might be better.